app/src/main/java/com/babylon/wallet/android/WalletApp.kt

```diff
@@ -59,140 +62,171 @@ fun WalletApp(
     val navController = rememberNavController()
     var showNotSecuredDialog by remember { mutableStateOf(false) }
     var showSecureFolderWarning by rememberSaveable { mutableStateOf(false) }
-    NavigationHost(
-        modifier = modifier.fillMaxSize(),
-        startDestination = MAIN_ROUTE,
-        navController = navController,
-        mainUiState = mainViewModel.state,
-        onCloseApp = onCloseApp
-    )
-    LaunchedEffect(Unit) {
-        mainViewModel.oneOffEvent.collect { event ->
-            when (event) {
-                is MainEvent.IncomingRequestEvent -> {
-                    if (event.request.needVerification) {
-                        navController.mobileConnect(event.request.interactionId)
-                        return@collect
-                    }
-                    when (val incomingRequest = event.request) {
-                        is IncomingMessage.IncomingRequest.TransactionRequest -> {
-                            navController.transactionReview(
-                                requestId = incomingRequest.interactionId
-                            )
+//    val owner = LocalLifecycleOwner.current
```

Comment:
Since the current implementation works great I'd recommend clearing up the unused code. If it might bring some value in the future maybe it's worth keeping it on a separate branch.